listenbrainz/stats/user.py

```diff
@@ -0,0 +1,176 @@
+# coding=utf-8
+#
+# listenbrainz-server - Server for the ListenBrainz project
+# Copyright (C) 2017 Param Singh
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import listenbrainz.config as config
+from listenbrainz import stats
+
+
+def get_top_recordings(musicbrainz_id, time_interval=None):
+    """ Get top recordings of user with given MusicBrainz ID over a particular time interval
+
+        Args:
+            musicbrainz_id (str): The MusicBrainz ID of the user
+            time_interval  (str): Interval in the BigQuery interval format
+                                  which can be seen here:
+                                  https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#timestamp_sub
+
+        Returns:
+            sorted list of the following format
+            [
+                {
+                    "track_name" (str)
+                    "recording_msid" (uuid)
+                    "artist_name" (str)
+                    "artist_msid" (uuid)
+                    "listen_count" (int)
+                }
+            ]
+    """
+
+    filter_clause = ""
+    if time_interval:
+        filter_clause = "AND listened_at > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL {})".format(time_interval)
```

Comment:
The interval is not defined anywhere that I can see. Should it be inclusive of the boundaries? (I think so).